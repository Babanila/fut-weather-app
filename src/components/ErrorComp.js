import React from "react";
import { ErrorCompStyles } from "../styles/ErrorCompStyles";

const ErrorComp = props => {
    return (
        <ErrorCompStyles>
            <p>
                Error:
                <span data-testid="errorMessage" className="error-message">
                    {props.data.message}
                </span>
            </p>
        </ErrorCompStyles>
    );
};

export default ErrorComp;
