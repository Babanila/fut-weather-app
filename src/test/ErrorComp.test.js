import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import ErrorComp from "../components/ErrorComp";

describe("ErrorComp Component", () => {
    let error;
    beforeEach(() => {
        error = { message: "There is an error" };
    });

    it("should renders without crashing", () => {
        const tree = renderer.create(<ErrorComp data={error} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should renders correct error message  correctly", () => {
        const { getByTestId, getByText } = render(<ErrorComp data={error} />);
        expect(getByTestId("errorMessage")).toHaveTextContent("There is an error");
        expect(getByText(error.message)).toHaveTextContent("There is an error");
    });
});
