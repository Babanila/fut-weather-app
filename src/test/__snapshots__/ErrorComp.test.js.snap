// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComp Component should renders without crashing 1`] = `
<div
  className="css-rgk63j"
>
  <p>
    Error:
    <span
      className="error-message"
      data-testid="errorMessage"
    >
      There is an error
    </span>
  </p>
</div>
`;
